sdks/go/pkg/beam/options/resource/hint.go

```diff
@@ -0,0 +1,194 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package resource supports Beam resource hints to specify scoped hints or annotations
+// to pipelines.
+//
+// See https://beam.apache.org/documentation/runtime/resource-hints/ for more information.
+package resource
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+
+	"github.com/dustin/go-humanize"
+)
+
+// Hints contains a list of hints for a given scope.
+type Hints struct {
+	h map[string]Hint
+}
+
+// MergeWithOuter produces a new list of Hints from this Hints, and the Hints from the outer scope.
```

Comment:
2 thoughts:
1) Could you specify the merge strategy? Looks like it is hint dependent, specifying that would be enough
2) Does I don't quite get the naming here - why is this MergeWithOuter and the analogous hint level operation is MergeWith? Personally, my vote would just be for Merge, but staying consistent is the more important piece.

Reply:
1. It is hint dependant, but there is a default approach. Both mentioned.
2. Switched to MergeWithOuter for the Hint as well. Because the merge is order dependant (outer scope vs inner scope) the order needs to be clear. Merge implies order independance. By default, an "inner" hint takes precedence. So the naming makes it clear how it's supposed to be called.

Reply:
Cool, as long as we're consistent that's good with me